### PR TITLE
save short orcids fixes #1060

### LINKED ISSUE #1060: User's ORCID ID to strip the URL prefix
In their NH "my account" page, users can provide a ORCID ID. The system expect it to be something like

```
0000-0000-0000-0000
```

but some users enter the URL:

```
https://orcid.org/0000-0000-0000-0000
```

There is no validation right now.

Ensure the form transforms the URL into the plain ID before saving it into `@user.meta[:orcid]` . Caveat: we use a helper for the form.

### PR DESCRIPTION
**Pull request by MontrealSergiy**
As some users save ORCID iD in the canonical, url form the stripping on save is added.
It is against the ORCID recommendations, but I guess not big deal.
On 'show' page I opted to render the full iD, following ORCID requirements.

### CODE REVIEW

File: BrainPortal/app/controllers/nh_users_controller.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ class NhUsersController < NeurohubApplicationController
 
   def show #:nodoc:
     @user = User.find(params[:id])
+    puts 'HELLO HELLO HELLO '
```
Comment (prioux) [inline BrainPortal/app/controllers/nh_users_controller.rb:32]: Please check your code diffs BEFORE you commit

**Review by prioux — CHANGES_REQUESTED** [review]
See my three comments.

File: BrainPortal/app/views/nh_users/_form.html.erb [hunk 2]
```diff
@@ -67,8 +67,10 @@
   </fieldset>
 
   <fieldset>
-    <%= f.label(:orcid, "Orcid Id") %>
+    <%= f.label(:orcid, "ORCID iD") %>
     <%= text_field_tag "meta[orcid]", @user.meta['orcid'], :size => 30 %><br />
+    <p class="field_note"><b>Note:</b> Enter <em>only</em> 16 digits of your ORCID iD with dashes between each four digits, for example 1234-5678-9012-3456, do not enter the url prefix.
```
[the remark this answers is not in the document]
Comment (prioux) [reply BrainPortal/app/views/nh_users/_form.html.erb:72]: Also, this message is no longer relevant if we allow the users to enter enything, isn't it? We extract the ID and present the full URL...

[the remark this answers is not in the document]
Comment (MontrealSergiy) [reply BrainPortal/app/views/nh_users/_form.html.erb:72]: That's just a recommendation to save user from unnecessary typing

**Comment by MontrealSergiy** [comment]
@prioux changes added. If you want we can add the validation (for distorted ORCIDs) UX.

File: BrainPortal/app/controllers/nh_users_controller.rb [hunk 3]
```diff
@@ -28,15 +28,16 @@ class NhUsersController < NeurohubApplicationController
   before_action :login_required
 
   def show #:nodoc:
-    @user = User.find(params[:id])
-    unless current_user.available_users.to_a.include?(@user)
+   @user = User.find(params[:id])
```
Comment (prioux) [inline BrainPortal/app/controllers/nh_users_controller.rb:31]: Why did you change up the indentation here?

File: BrainPortal/app/views/nh_users/_form.html.erb [hunk 4]
```diff
@@ -67,8 +67,10 @@
   </fieldset>
 
   <fieldset>
-    <%= f.label(:orcid, "Orcid Id") %>
+    <%= f.label(:orcid, "ORCID iD") %>
     <%= text_field_tag "meta[orcid]", @user.meta['orcid'], :size => 30 %><br />
+    <p class="field_note"><b>Note:</b> The url prefix is not required just enter all the 16 last symbols of your ORCID iD with dashes between each group of four, for example, 1234-5678-9098-765X
```
Comment (prioux) [inline BrainPortal/app/views/nh_users/_form.html.erb:72]: Text change

```
You can enter the full ORCID URI, whichs starts with <em>https://</em>, or just the numerical identifier, e.g. <em>1234-5678-9012-345X</em>
````

Check the em or strong tag work (maybe Candice's template cancels their style here)

Comment (prioux) [reply BrainPortal/app/views/nh_users/_form.html.erb:72]: Also in the text_field_tag (previous unchanged line) we shoudl not invoke your orcid_canonize method to show the full URI in the text box

**Review by prioux — CHANGES_REQUESTED** [review]
See comments

File: BrainPortal/app/views/nh_users/_form.html.erb (hunk 4 above)
Comment (MontrealSergiy) [reply BrainPortal/app/views/nh_users/_form.html.erb:72]: @prioux I would say 'the numerical part' or numerical 'numerical identifier (with dashes)' - dashes are required

Comment (prioux) [reply BrainPortal/app/views/nh_users/_form.html.erb:72]: Sure. Fine with me. As long as it's clear.

**Comment by MontrealSergiy** [comment]
@prioux done, using <strong> instead of   <em>  since 'filed explanation' is already italic
